code/datums/datacore.dm

```diff
@@ -302,11 +302,18 @@ var/global/list/PDA_Manifest = list()
 	var/list/all_jobs = get_job_datums()
 
 	for(var/datum/job/J in all_jobs)
-		var/list/alttitles = get_alternate_titles(J.title)
-		if(!J)	continue
-		if(assignment in alttitles)
-			real_title = J.title
+		if(J.title == rank)					//If we have a rank, just default to using that.
+			real_title = rank
 			break
+		else if(J.title == assignment)
+			real_title = rank
+			break
+		else
+			var/list/alttitles = get_alternate_titles(J.title)
+			if(!J)	continue

```

Comment:
Why do you check `J` after the point where a falsey value in `J` would have runtimed?

Reply:
Thats the part I did not modify. I have not a single clue why it was like that in the first place, but it wasn't my place to judge.

Reply:
You can safely delete it. `J` is implicitly typechecked by the for loop.

Reply:
Sure. I was wandering why it was like that when its basically a check for an always false thing, but I decided that I'm a bit too sleepy to try and delete extremely ambigous-looking things that were there before me.